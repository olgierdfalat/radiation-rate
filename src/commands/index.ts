export * from './checksum';
export * from './filterFiles';
export * from './wrongFiles';
export * from './excelExport';
