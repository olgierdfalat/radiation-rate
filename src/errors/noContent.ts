export class NoContent extends Error {
  constructor(message?: string) {
      super(message || 'No content'); // 'Error' breaks prototype chain here
      Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
  }
}