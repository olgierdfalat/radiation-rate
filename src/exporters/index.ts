export * from './worksheetsDataProvider';
export * from './excel';