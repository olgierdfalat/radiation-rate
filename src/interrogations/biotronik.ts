import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class Biotronik extends Interrogation {
  async getData(): Promise<parsers.BiotronikData> {
    const content = await this.getContent();
    return new parsers.BiotronikData(content, this.filePath);
  }
}