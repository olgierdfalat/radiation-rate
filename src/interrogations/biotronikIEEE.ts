import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class BiotronikIEEE extends Interrogation {
  async getData(): Promise<parsers.BiotronikIEEEData> {
    const content = await this.getContent();
    return new parsers.BiotronikIEEEData(content, this.filePath);
  }
}