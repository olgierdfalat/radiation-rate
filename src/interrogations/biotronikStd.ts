import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class BiotronikStd extends Interrogation {
  async getData(): Promise<parsers.BiotronikStdData> {
    const content = await this.getContent();
    return new parsers.BiotronikStdData(content, this.filePath);
  }
}