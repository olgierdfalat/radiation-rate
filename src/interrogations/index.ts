export * from './interrogation';
export * from './stJude';
export * from './biotronikStd';
export * from '../exporters/worksheetsDataProvider';
export * from './interrogationFactory';