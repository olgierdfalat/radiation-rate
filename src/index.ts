import { Command } from 'commander';
import * as commands from './commands';


const command = new Command();
command
  .version('0.0.1')
  .description('CLI for converting heart valves output data into Excel format')
  .option('-c, --checksum <device>', 'generates checksums of all log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.checksum)
  .option('-e, --excluded <device>', 'generates list of excluded log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.excludedFiles)
  .option('-i, --included <device>', 'generates list of included log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.includedFiles)
  .option('-w, --invalid <device>', 'generates list of wrong log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.invalidFiles)
  .parse(process.argv);

