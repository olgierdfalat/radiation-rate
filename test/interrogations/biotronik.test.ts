import * as path from 'path';
import * as interrogations from '../../src/interrogations';
import * as errors from '../../src/errors';


describe('interrogation Biotronik class', () => {
  it('should get structured data row from file', async () => {
    const filePath = path.join(__dirname, '../fixtures/Biotronik/BIO_2016-12-23_12-12-56_ANONYMOUS_A_60485754.xml');
    const biotronik = new interrogations.Biotronik(filePath);
    const data = await biotronik.getData();
    expect(data.getRow()).toMatchSnapshot();
  });

  it('should throw No Content exception for empty log file', async () => {
    const filePath = path.join(__dirname, '../fixtures/empty-invalid.log');
    const biotronik = new interrogations.Biotronik(filePath);
    await expect(biotronik.getData()).rejects.toThrow(errors.NoContent);
  });
});