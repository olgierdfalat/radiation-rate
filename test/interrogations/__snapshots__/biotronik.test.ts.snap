// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interrogation Biotronik class should get structured data row from file 1`] = `
Object {
  "birthDate": "1949-07-10",
  "commentary": "Symptom=Prophylactic indication, Aetiology=Dilative cardiomyopathy (ischemic), ECGIndication=QRS widening",
  "generationDate": "2016-12-23T12:15:28",
  "icdClinicDate": "2016-12-23T12:12:00",
  "icdClinicEvaluationBradyProgramming_AMS_FallbackRate_bpm": "DDI",
  "icdClinicEvaluationBradyProgramming_AMS_Rate_bpm": "160",
  "icdClinicEvaluationBradyProgramming_AVDelay_AdaptiveAVDelayStatus": "ON",
  "icdClinicEvaluationBradyProgramming_AVDelay_AdaptivePacedMinimum_ms": "220",
  "icdClinicEvaluationBradyProgramming_AVDelay_AdaptiveRateMaximum_bpm": "130",
  "icdClinicEvaluationBradyProgramming_AVDelay_AdaptiveRateMinimum_bpm": "60",
  "icdClinicEvaluationBradyProgramming_AVDelay_AdaptiveSensedMinimum_ms": "190",
  "icdClinicEvaluationBradyProgramming_AVDelay_Paced_ms": "240",
  "icdClinicEvaluationBradyProgramming_AVDelay_Sensed_ms": "210",
  "icdClinicEvaluationBradyProgramming_AVDelay_VVDelay_ms": "",
  "icdClinicEvaluationBradyProgramming_HysteresisRate_bpm": "60",
  "icdClinicEvaluationBradyProgramming_LowerRate_bpm": "60",
  "icdClinicEvaluationBradyProgramming_Notes": "MSW.Rate =70bpm; MSw mode=DDI;",
  "icdClinicEvaluationBradyProgramming_PMTIntervention": "ON",
  "icdClinicEvaluationBradyProgramming_PVCIntervention": "475",
  "icdClinicEvaluationBradyProgramming_PacingMode": "DDD",
  "icdClinicEvaluationBradyProgramming_Pacing_Chamber_RA": "RA",
  "icdClinicEvaluationBradyProgramming_Pacing_Chamber_RA_Amplitude_volts": "1.5",
  "icdClinicEvaluationBradyProgramming_Pacing_Chamber_RA_Polarity": "B",
  "icdClinicEvaluationBradyProgramming_Pacing_Chamber_RA_PulseWidth_ms": "0.5",
  "icdClinicEvaluationBradyProgramming_Pacing_Chamber_RV": "RV",
  "icdClinicEvaluationBradyProgramming_Pacing_Chamber_RV_Amplitude_volts": "2.2",
  "icdClinicEvaluationBradyProgramming_Pacing_Chamber_RV_Polarity": "B",
  "icdClinicEvaluationBradyProgramming_Pacing_Chamber_RV_PulseWidth_ms": "0.5",
  "icdClinicEvaluationBradyProgramming_Sensing_Chamber_RA": "RA",
  "icdClinicEvaluationBradyProgramming_Sensing_Chamber_RA_Amplitude_millivolts": "0.4",
  "icdClinicEvaluationBradyProgramming_Sensing_Chamber_RA_Polarity": "B",
  "icdClinicEvaluationBradyProgramming_Sensing_Chamber_RV": "RV",
  "icdClinicEvaluationBradyProgramming_Sensing_Chamber_RV_Amplitude_millivolts": "0.8",
  "icdClinicEvaluationBradyProgramming_Sensing_Chamber_RV_BlankingPeriod_ms": "AUTO",
  "icdClinicEvaluationBradyProgramming_Sensing_Chamber_RV_Polarity": "B",
  "icdClinicEvaluationBradyProgramming_TrackingRate_bpm": "130",
  "icdClinicEvaluation_ICDTelemetry_BatteryVoltage_volts": "2.85",
  "icdClinicEvaluation_ICDTelemetry_LastHighVoltageChargeTime_seconds": "16.6",
  "icdClinicEvaluation_ICDTelemetry_LastHighVoltageEnergy": "40",
  "icdClinicEvaluation_ICDTelemetry_LeadImpedanceAtrial": "",
  "icdClinicEvaluation_ICDTelemetry_LeadImpedanceHighVoltage": "",
  "icdClinicEvaluation_ICDTelemetry_LeadImpedanceLeftVentricular": "",
  "icdClinicEvaluation_ICDTelemetry_LeadImpedanceRightVentricular": "",
  "implantDate": "2010-10-25",
  "institutionName": "BIOTRONIK SE & Co. KG",
  "manufacturer": "Biotronik",
  "model": "Lumax 340 DR-T",
  "nameFirst": "ANONYMOUS",
  "nameLast": "ANONYMOUS",
  "schemaVersion": "2.0",
  "serialNumber": "60485754",
}
`;
